File: model/github_queries.py


class QueryType(Enum):
    GetLastEvent = 0
    GetEventsPage = 1

class ResponseStatus(Enum):
    OK = 0
    KO = 1

class ApiResponse():
  def __init__(self, *args, **kwargs):
    return super().__init__(*args, **kwargs)

File: model/github_queries.go
package model

import (
	"github.com/google/go-github/v24/github"
)

type ApiQuery struct {
	Type          QueryType          `json:"type"`
	ID            string             `json:"id"`
	ListOptions   github.ListOptions `json:"options"`
	ResponseTopic string             `json:"response_topic"`
}

type QueryType int

const (
	GetLastEvents QueryType = iota
	GetEventsPage
)

type ResponseStatus int

const (
	OK ResponseStatus = iota
	KO
)

type ApiResponse struct {
	ID      string         `json:"id"`
	Status  ResponseStatus `json:"status"`
	Payload []byte         `json:"payload"`
}
